packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,375 @@
+          Total
+        </TableFooterCell>
+        <TableFooterCell>
+          {computedTotalBalance}
+        </TableFooterCell>
+      </TableFooterRow>
+    </TableFooter>
+
+    <TablePagination 
+      navigationType='compact' 
+      label='1-10 of 100'
+      labelPosition='left'
+      showRowCountPicker
+      rowCount='25'
+      onPaginationChange={console.log}
+      onRowCountPickerChange={console.log}
+      /> 
+   </>
+  )
+}
+</Table>
+```
+
+##### Pros
+- Composable API is more intuitive and resembles the native HTML structure of a table
+- It is easier to understand since the API design is similar to rest of the Blade components
+- It is easier to extend and add new feature to individual table components
+- We could leverage tree-shaking to only import the components that are being used in the consumer's table
+
+##### Cons
+- Composable API is more verbose and requires more effort to create a table by composing together multiple components
+- Most popular table libraries like AG Grid, React Table, etc. use a compact API which is battle tested and developers are more familiar with the same
+
+#### 2. Compact API
+Compact API is a pattern where we have a single component that takes in all the data and renders a table. This approach is more concise and requires less effort to create a table. However, it is also less intuitive and does not resemble the native HTML structure of a table.
+
+```jsx
+// Mock data & functions
+const data = [{
+    firstName: 'John',
+    lastName: 'Doe',
+    balance: 1000
+},
+{
+    firstName: 'Jane',
+    lastName: 'Doe',
+    balance: 2000
+}];
+
+
+const sortFunctions = {
+  firstName: (array) => array.sort((a, b) => a.name.localeCompare(b.name)),
+};
+
+const onSelectionChange = (selectedItems) => {
+  console.log(selectedItems);
+};
+
+const columnDefinition = [
+    {
+        header: 'First Name',
+        headerKey: 'firstName',
+        cell: (row) => row.firstName,
+        isSticky: true,
+        columnStart: undefined,
+        columnEnd: undefined,
+    },
+    {
+        header: 'Last Name',
+        headerKey: 'lastName',
+        cell: (row) => row.lastName,
+        isSticky: true,
+        columnStart: undefined,
+        columnEnd: undefined,
+    },
+    {
+        header: () => (
+            <>
+                Balance
+                <Tooltip content='Current bank balance'>
+                    <InfoIcon size='medium' />
+                </Tooltip>
+            </>),
+        headerKey: 'balance',
+        cell: (row) => (
+            <>
+                {row.balance}
+                <Badge>Low</Badge>
+            </>),
+        isSticky: true,
+        columnStart: undefined,
+        columnEnd: undefined,
+    }
+];
+
+const footerDefinition = [
+    {
+        footer: 'Total',
+        columnStart: 1,
+        columnEnd: 4,
+    },
+    {
+        footer: computedTotalBalance,
+    }
+];
+
+const pagination = {
+  showRowCountPicker?: boolean
```

Comment:
what is row count picker?

Reply:
It decides how many rows should be shown per page
<img width="145" alt="Screenshot 2023-10-25 at 5 19 27 PM" src="https://github.com/razorpay/blade/assets/24487274/f7a29f31-9945-4fc0-a10c-e7f7d2c569f6">